### PR TITLE
Admin Employee and rspec

### PR DESCRIPTION
**Pull request by Namrata12gupta**

### CODE REVIEW

File: app/controllers/admin_main/employees_controller.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+module AdminMain
+  class EmployeesController < ApplicationController
+    def index
+      redirect_to new_admin_main_employee_path
+    end
+
+    def new
+      @employee = Employee.new
+      @designations = Designation.all
```
Comment (rorpyll) [inline app/controllers/admin_main/employees_controller.rb:11]: use class method for similar code

File: app/controllers/admin_main/employees_controller.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      @employee = Employee.new(employee_params)
+      if @employee.save
+        redirect_to @employee
+      else
+        render :new
+      end
+    end
+
+    def edit
+      @employee = Employee.find_by_id params[:id]
```
Comment (rorpyll) [inline app/controllers/admin_main/employees_controller.rb:29]: use class method for similar code

**Review by rorpyll — CHANGES_REQUESTED** [review]

File: app/views/shared/_header.html.erb [hunk 3]
```diff
@@ -123,18 +131,35 @@
...
         
       </li>
+      
       </li class="nav-item">
+          <% if current_employee.role.name == "Employee" || current_employee.role.name == "HR" %>
           <%= link_to  holidays_path , class: "nav-link collapsed" do %>
           <i class="bi bi-menu-button-wide"></i>
           <%= 'Holidays' %>
           <% end %>
+          <% else %>
```
Comment (ror-uday) [inline app/views/shared/_header.html.erb:141]: correct indentation

File: config/routes.rb [hunk 4]
```diff
@@ -35,10 +35,18 @@
         end
       end
     end
+
     resources :attendences
     get '/emp-attendance/:id', to: 'attendences#show_attendence', as: 'show_attendance'
     get '/search', to: 'attendences#search'
   end
+
+  namespace :admin_main do
```
Comment (ror-uday) [inline config/routes.rb:44]: use rubocop

**Review by ror-uday — CHANGES_REQUESTED** [review]

File: app/controllers/admin_main/employees_controller.rb (hunk 1 above)
Comment (Namrata12gupta) [reply app/controllers/admin_main/employees_controller.rb]: Changes Done

File: app/controllers/admin_main/employees_controller.rb (hunk 2 above)
Comment (Namrata12gupta) [reply app/controllers/admin_main/employees_controller.rb]: Changes Done

File: app/views/shared/_header.html.erb [hunk 5]
```diff
@@ -84,18 +84,26 @@
...
+            <%= 'Profile' %>
+        <% end %>
+        <% end %>
+      </li>
       
       <li class="nav-item">
           <%= link_to employees_path , class: "nav-link collapsed" do %>
           <i class="bi bi-person"></i>
-          <% unless current_employee.role.name == "HR" %>
+          <% unless current_employee.role.name == "HR" ||current_employee.role.name == "Admin" %>
```
[the remark this answers is not in the document]
Comment (Namrata12gupta) [reply app/views/shared/_header.html.erb]: Changes Done

File: app/models/ability.rb [hunk 6]
```diff
@@ -6,7 +6,7 @@ class Ability
   def initialize(employee)
     # Define abilities for the passed in user here. For example:
     #   user ||= User.new # guest user (not logged in)
-        if employee.role.name == "HR"
+        if employee.role.name == "HR" || employee.role.name == "Admin"
```
Comment (rorpyll) [inline app/models/ability.rb:9]: why are you not use instance method

File: spec/controllers/admin_main/employee_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    it 'return nil when employee not exist' do
+      patch :update, params: { employee: { name: 'Xyz' }, id: '' }
+      expect(assigns(:employee)).to eq(nil)
+    end
+  end
+
+  describe '#destroy' do
+    it 'delete employee data' do
+      delete :destroy, params: { id: employee.id }
+      expect(response.status).to eq(302)
```
Comment (rorpyll) [inline spec/controllers/admin_main/employee_controller_spec.rb:56]: conspire with object like employee count  2 after destroy employee count should be 1

**Review by rorpyll — CHANGES_REQUESTED** [review]

**Review by rorpyll — APPROVED** [review]

**Review by ror-uday — APPROVED** [review]

File: app/controllers/admin_main/employees_controller.rb [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+  class EmployeesController < ApplicationController
+    before_action :set_employee, only: %i[edit update destroy]
+
+    def new
+      @employee = Employee.new
+      set_designations
+    end
+
+    def create
+      set_designations
```
Comment (salvimohan) [inline app/controllers/admin_main/employees_controller.rb:13]: create filter for it

File: app/views/employees/index.html.erb [hunk 9]
```diff
@@ -372,6 +384,17 @@
                       <%= link_to 'Details', employee, class: "btn btn-primary"%>
                       <% end %>
                     </td>
+
+                    <td><% if current_employee.is_admin? %>
```
Comment (salvimohan) [inline app/views/employees/index.html.erb:388]: <% if current_employee.is_admin? %>
 <td>  <%= link_to 'Edit', edit_admin_main_employee_path(employee), class: "btn btn-primary"%></td>
 <td><%= link_to 'Destroy', admin_main_employee_path(employee),method: :delete, data: { confirm: 'Are you sure?' }, class: "btn btn-primary"%>     </td>
 <% end %>

File: app/views/shared/_header.html.erb [hunk 10]
```diff
@@ -24,55 +23,42 @@
...
   <aside id="sidebar" class="sidebar">
-
     <ul class="sidebar-nav" id="sidebar-nav">
 
       <li class="nav-item">
         <%= link_to root_path, class: "nav-link collapsed a" do %> 
-          <i class="bi bi-grid"></i>
-          <%= "Dashboard" %>
+        <i class="bi bi-grid"></i>
+        <%= "Dashboard" %>
```
Comment (salvimohan) [inline app/views/shared/_header.html.erb:61]: why

**Pull request merged**